project_task_autocreate/models/project_task_rule_mixin.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ProjectTaskRuleMixin(models.AbstractModel):
+    """Task Creation Mixin"""
+    _name = 'project.task.rule.mixin'
+    _description = __doc__
+
+    _has_on_state_rules = False
+    _has_on_write_rules = False
+    _has_on_create_rules = False
+    _has_on_delete_rules = False
+
+    template_project_id = fields.Many2one(
+        string='Project For Tasks From Templates',
+        comodel_name='project.project',
+    )
+
+    def _register_hook(self):
+        super(ProjectTaskRuleMixin, self)._register_hook()
+        model = self._name
+        if model != 'project.task.rule.mixin':
+            self.env['project.task.rule'].register_model(model)
+            self._update_rule_hooks(self.env)
+
+    @api.model
+    def _add_field(self, name, field):
+        super(ProjectTaskRuleMixin, self)._add_field(name, field)
+        model = self._name
+        if (name == 'state' and model != 'project.task.rule.mixin' and
+                field.type == 'selection'):
+            self._update_ir_model_state(model, self._fields['state'].selection)
+
+    @classmethod
+    def _update_rule_hooks(cls, env):
+        rules = env['project.task.rule']
+        cls._has_on_create_rules = bool(rules.search([
+            ('model', '=', cls._name),
+            ('on_create', '=', True),
+        ]))
+        cls._has_on_write_rules = bool(rules.search([
+            ('model', '=', cls._name),
+            ('on_write', '=', True),
+        ]))
+        cls._has_on_state_rules = bool(rules.search([
+            ('model', '=', cls._name),
+            ('on_state', '=', True),
+        ]))
+        cls._has_on_delete_rules = bool(rules.search([
+            ('model', '=', cls._name),
+            ('on_delete', '=', True),
+        ]))
```

Comment:
@dufresnedavid that's true, then again once I remove the calls to _update_rule_hooks in project.task.rule this will only be called once at worker startup